assets/src/blocks/Covers/CoversFrontend.js

```diff
@@ -58,14 +58,27 @@ export const CoversFrontend = attributes => {
       const initialScrollPosition = covers.scrollLeft;
       covers.scrollLeft = initialScrollPosition + (direction === 'next' ? scrollOffset : -scrollOffset);
       setRow(direction === 'next' ? row + 1 : row - 1);
-    } else if (rowNumber && rowNumber !== row) {
+    } else if (rowNumber) {
       covers.scrollLeft = (rowNumber - 1) * scrollOffset;
       setRow(rowNumber);
     }
   };
 
+  const onResizeHandler = () => {
+    slideCovers('', 1);
+  }
+
   const showLoadMoreButton = !isCarouselLayout && !!initialRowsLimit && covers.length > (amountOfCoversPerRow * row);
 
+  useEffect(() => {
```

Comment:
What do you think to implement [this other approach](https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/778/files#diff-93c303aba0166eba7362c8c44643bb67e809e03106503a13337b8e00ce45a7a5R70-R82) added to the `useEffect`? It would be probably better since you're running all the resizing logic only if it's a carousel layout and the handler method is being created once. 

If you compare with this new implementation, on one hand you're checking if it's a carousel layout for adding the `resize` event handler and on the other hand you're creating the resize the `onResizeHandler` no matter if it's listening added the `resize` handler.

Does it make sense?

Reply:
Yes, it makes sense! Thanks, I've updated the PR 🙂